contracts/voting-verifier/src/events.rs

```diff
@@ -175,24 +191,28 @@ fn parse_message_id(
 
 impl VerifierSetConfirmation {
     pub fn new(
-        message_id: &str,
+        message_id: nonempty::String,
         msg_id_format: MessageIdFormat,
         verifier_set: VerifierSet,
     ) -> Result<Self, ContractError> {
-        let (tx_id, event_index) = parse_message_id(message_id, &msg_id_format)?;
+        let (tx_id, event_index) = parse_message_id(&message_id, &msg_id_format)?;
 
         Ok(Self {
             tx_id,
             event_index,
+            message_id,
             verifier_set,
         })
     }
 }
 
 #[cw_serde]
 pub struct TxEventConfirmation {
+    #[deprecated(since = "1.1.0", note = "use message_id field instead")]
     pub tx_id: nonempty::String,
+    #[deprecated(since = "1.1.0", note = "use message_id field instead")]
```

Comment:
```suggestion
    #[deprecated(since = "1.3.0", note = "use message_id field instead")]
    pub tx_id: nonempty::String,
    #[deprecated(since = "1.3.0", note = "use message_id field instead")]
```